packages/preset-uno/src/variants/mix.ts

```diff
@@ -0,0 +1,54 @@
+import type { CSSColorValue, Variant } from '@unocss/core'
+import { parseCssColor } from '@unocss/preset-mini/utils'
+
+const mixComponent = (v1: string | number, v2: string | number, w: string | number) => `calc(${v2} + (${v1} - ${v2}) * ${w} / 100)`
+
+const mixColor = (color1: string | CSSColorValue, color2: string | CSSColorValue, weight: string | number): CSSColorValue | undefined => {
+  const colors = [color1, color2]
+  const cssColors: CSSColorValue[] = []
+  for (let c = 0; c < 2; ++c) {
+    const color = (typeof colors[c] === 'string' ? parseCssColor(colors[c] as string) : colors[c]) as CSSColorValue | undefined
+    if (!color || !['rgb', 'rgba'].includes(color.type))
+      return
+    cssColors.push(color)
+  }
+
+  const newComponents = []
+  for (let x = 0; x < 3; ++x)
+    newComponents.push(mixComponent(cssColors[0].components[x], cssColors[1].components[x], weight))
+
+  return {
+    type: 'rgb',
+    components: newComponents,
+    alpha: mixComponent(cssColors[0].alpha ?? 1, cssColors[1].alpha ?? 1, weight),
+  }
+}
+
+const tint = (color: string | CSSColorValue, weight: string | number) => mixColor('#fff', color, weight)
+const shade = (color: string | CSSColorValue, weight: string | number) => mixColor('#000', color, weight)
+const shift = (color: string | CSSColorValue, weight: string | number) => parseInt(`${weight}`, 10) > 0 ? shade(color, weight) : tint(color, `-${weight}`)
+const fns: Record<string, (color: string | CSSColorValue, weight: string | number) => CSSColorValue | undefined> = { tint, shade, shift }
+
+export const variantColorMix: Variant = (matcher) => {
```

Comment:
Can you add a short JSDoc here to describe how this util work? This could help ppl to read the code and find the usage.